system/BaseModel.php

```diff
@@ -737,6 +742,9 @@ public function insert($data = null, bool $returnID = true)
 
         $result = $this->doInsert($eventData['data']);
 
+        // Reset $prohibitInsertEmpty value.
+        $this->prohibitInsertEmpty = true;
```

Comment:
I think this is too forceful. If a developer sets the property and then it gets reset here that would feel very buggy. I think the workaround is to implement a temporary value, like how `$tempAllowCallbacks`, etc, work.

Reply:
I removed it and now `$allowEmptyInserts` is a normal property.